internal/ingress/controller/parser/kongstate/consumer.go

```diff
@@ -13,17 +13,40 @@ import (
 type Consumer struct {
 	kong.Consumer
 	Plugins    []kong.Plugin
-	KeyAuths   []*kong.KeyAuth
-	HMACAuths  []*kong.HMACAuth
-	JWTAuths   []*kong.JWTAuth
-	BasicAuths []*kong.BasicAuth
+	KeyAuths   map[string]*kong.KeyAuth
+	HMACAuths  map[string]*kong.HMACAuth
+	JWTAuths   map[string]*kong.JWTAuth
+	BasicAuths map[string]*kong.BasicAuth
 	ACLGroups  []*kong.ACLGroup
 
-	Oauth2Creds []*kong.Oauth2Credential
+	Oauth2Creds map[string]*kong.Oauth2Credential
 
 	K8sKongConsumer configurationv1.KongConsumer
 }
 
+func NewConsumer() Consumer {
+	return Consumer{}.initEmpty()
+}
+
+func (c Consumer) initEmpty() Consumer {
```

Comment:
question: I'm simply curious for my own learning what instigated this change? I would assume this is for convenience to avoid having to initialize these manually every time you make a new `Consumer`, is that pretty much it?

And that being the case, we should perhaps add some documentation for the `NewConsumer()` public function which will point out this behavior for readers of our Godoc.

Reply:
Before this PR, a zero-initialized `Consumer` was a valid consumer. Now it is not because of introduction of `map`s. Hence `NewConsumer()`. `NewConsumer` could simply initialize all fields of `Consumer`, but that would be tedious for tests. In tests we often need "a minimal consumer with one specific map initialized, and the rest left empty".

This is an idiom I've been following in such scenario, but for my own learning I'd love to see how others deal with this problem (of partial initialization for testing purposes).